storage/storage_interface.ml

```diff
@@ -13,7 +13,63 @@
  *)
 (**
  * @group Storage
- *)
+*)
+open Rpc
+open Idl
+
+module D = Debug.Make (struct
+    let name = "storage_interface"
+  end)
+open D
+
+type rpc_t = Rpc.t
+
+let typ_of_rpc_t =
+  let open Types in
+  Abstract
+    { aname= "Rpc.t"
+    ; test_data= [Null]
+    ; rpc_of= (fun x -> x)
+    ; of_rpc= (fun x -> Ok x) }
+
+
+
+module TypeCombinators = struct
+  let option ?name ?(description= []) d =
+    let open Rpc.Types in
+    let name =
+      match name with Some n -> n | None -> Printf.sprintf "%s option" d.name
+    in
+    {name; description; ty= Option d.ty}
+
+  let list ?name ?(description= []) d =
+    let open Rpc.Types in
+    let name =
+      match name with
+      | Some n -> n
+      | None -> Printf.sprintf "list of %ss" d.name
+    in
+    {name; description; ty= List d.ty}
+
+  let pair ?name ?(description= []) (p1, p2) =
+    let open Rpc.Types in
+    let name =
+      match name with
+      | Some n -> n
+      | None -> Printf.sprintf "pair of %s and %s" p1.name p2.name
+    in
+    {name; description; ty= Tuple (p1.ty, p2.ty)}
+
+  let triple ?name ?(description= []) (p1, p2, p3) =
+    let open Rpc.Types in
+    let name =
+      match name with
+      | Some n -> n
+      | None ->
+        Printf.sprintf "triple of %s, %s and %s" p1.name p2.name p3.name
+    in
+    {name; description; ty= Tuple3 (p1.ty, p2.ty, p3.ty)}
+end
```

Comment:
Are these type combinators more generally useful, should we put them in some shared module in xcp-idl?

Reply:
Yes, I've got them in the xen interface too. I was wanting to update the ocaml-rpc library with these (and more) but thought we could do that later.